### PR TITLE
`Dockerfile`s: Remove cache mount-type

### LINKED ISSUE #379: `Dockerfile`s: Remove cache mount-type
**Describe the enhancement you'd like to see**

Remove unneeded caching:

```Dockerfile
...
RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/root/go/pkg ...
...
```

Etc. from `Dockerfile`s:

* `build/agent/Dockerfile`.
* `build/agent/Dockerfile-debug`.
* `build/manager/Dockerfile`.
* `build/agent/Dockerfile-debug`.

**Additional context**

With the introduction of vendoring, please see kubeshop/kusk#32, we will not be downloading the dependencies each time so caching really isn't needed here and it also complicates the build.

### PR DESCRIPTION
**Pull request by mbana**
Closes #379.

`Dockerfile`s:
--------------

1. Use `docker.io/golang:1.17`, i.e., fully qualified path to image.
2. Remove unneeded caching:

```Dockerfile
...
RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/root/go/pkg ...
...
```

Etc. from `Dockerfile`s:

* `build/agent/Dockerfile`.
* `build/agent/Dockerfile-debug`.
* `build/manager/Dockerfile`.
* `build/agent/Dockerfile-debug`.

`Makefile`
----------

1. Use `docker.io/golang:1.17`, i.e., fully qualified path to image.
2. Tidyup `Makefile`.
3. Add `--profile kgw` to `docker-images-cache` target.

Misc
----

* Ignore `.tmp/` folder.

This PR...

---

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: build/manager/Dockerfile-debug [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.17 as builder
+FROM docker.io/golang:1.17 as builder
```
Comment (kylehodgetts) [inline build/manager/Dockerfile-debug:2]: Why is this needed?

Comment (mbana) [reply build/manager/Dockerfile-debug:2]: I get this prompt, which is undesirable, on Linux when I use `podman` instead of `docker`:

```sh
$ make docker-build-agent             
podman build --no-cache -t kusk-gateway-agent:dev --build-arg GOPROXY= -f ./build/agent/Dockerfile .
WARN[0000] missing "TELEMETRY_TOKEN" build argument. Try adding "--build-arg TELEMETRY_TOKEN=<VALUE>" to the command line 
[1/2] STEP 1/10: FROM golang:1.17 AS builder
? Please select an image: 
  ▸ registry.fedoraproject.org/golang:1.17
    registry.access.redhat.com/golang:1.17
    docker.io/library/golang:1.17
    quay.io/golang:1.17
```

As you can see, in the above, it finds the image in multiple registries which leads to a prompt.

In general, I think it's better to fully qualify the image name/path.

Comment (kylehodgetts) [reply build/manager/Dockerfile-debug:2]: Ah, i didn't know about this, given Ive never used Podman

**Review by kylehodgetts — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]
